### PR TITLE
nitdoc: fix things, add more tests and benches

### PR DESCRIPTION
**Pull request by Morriar**
Two fixes:

* no more multiple instanciation of ToolContext (-25% of total nitdoc time)
* fixed JS scripts (no more bug with the search bar that does not display)

Testing:
* add more tests on pages structure
* add a script to build the documentation from origin/master version (easier to diff outputed doc after local changes)
* add a benchmark to nitdoc against the origin/master version

Performances comparison after fix on ToolContext:

image

(`CI-nitdoc` is busy right now on Jenkins so you will have to wait for the demo)

### CODE REVIEW

**Comment by privat** [comment]
+1

**Comment by Morriar** [comment]
Reroll with @privat remarks

**Comment by jpages** [comment]
+1, especially for the research bar !

**Comment by Morriar** [comment]
Reroll with the correct line ending in js files

**Comment by Morriar** [comment]
ok to test

**Comment by Morriar** [comment]
Hum... The test failing is because a section is not at the same position in the tree when generated from nitg-s or nitg-se.

I tried on my laptop: It works with all engines.

I tried the same thing on pratchett: Also works with all engines.

Example with nitg-s on pratchett:
~~~sh
me@pratchett $ ./tests.sh --engine nitg-s ../src/nitdoc.nit
Found binary for engine nitg-s: ../bin/nitc --separate  --compile-dir .nit_compile
=> nitdoc: . [ok] out/nitdoc.res sav//nitdoc.res
==> nitdoc args 1 [ok] out/nitdoc_args1.res sav//nitdoc_args1.res
==> nitdoc args 2 [ok] out/nitdoc_args2.res sav//nitdoc_args2.res
==> nitdoc args 3 [ok] out/nitdoc_args3.res sav//nitdoc_args3.res
engine: nitg-s (nitc --separate  --compile-dir .nit_compile)
ok:  4 / 4
~~~

I'm not sure of what is really compiled by jenkins since the [change list](http://gresil.org/jenkins/job/CI_github/2943/changes) doesn't want to show up. Any idea on that?

File: share/man/nitdoc.md [hunk 1]
```diff
@@ -108,6 +108,11 @@ See niunit(1) for details.
 `--piwik-site-id`
 :   Piwik site ID
 
+## TESTING
+
+`--test`
+:   Only print the pages structure. Do not generate nothing.
```
Comment (privat) [inline share/man/nitdoc.md]: too much negations

**Comment by privat** [comment]
please update sav/nitdoc_args4.res so I can merge

**Pull request merged**